ggml-alloc.c

```diff
@@ -167,12 +167,11 @@ void ggml_allocr_alloc(struct ggml_allocr * alloc, struct ggml_tensor * tensor)
         struct free_block * block = &alloc->free_blocks[alloc->n_free_blocks - 1];
         if (block->size >= size) {
             best_fit_block = alloc->n_free_blocks - 1;
-            max_avail = MAX(max_avail, block->size);
```

Comment:
The correct solution is to move the assignment before the if block. @xaedes already fixed this in https://github.com/ggerganov/llama.cpp/pull/2632